src/object.d

```diff
@@ -4956,3 +4956,315 @@ string _d_assert_fail(string comp, A, B)(A a, B b) @nogc @safe nothrow pure
     buffer[n .. n + valB.length] = valB;
     return (() @trusted => cast(string) buffer)();
 }
+
+// These are needed to fake the safety level to `@safe pure nothrow`.
+// Without this code will break as these hooks where previously exported
+// as having this safety level.
+T[] _d_arrayappendT(T)(ref T[] x, scope T[] y) @trusted
```

Comment:
Do we need to return `T[]`?

Reply:
Is the `@trusted` on this declaration redundant?  It seems that since the call to `_d_arrayappendT_` is wrapped `@trusted.` the  `@trusted` attribute at entry point is not necessary.

Reply:
> Do we need to return T[]?

For chaining?

Reply:
It returns `T[]` because the original hook returned `void[]`. I also just found a problem with the current implementation in dmd:
```d
int[] arr;
((arr ~= [1,2,3]) ~= 4) ~= 5;
```
This crashes in the backend for dmd (https://run.dlang.io/is/UZAIuX), while working as expected in ldc (`arr == [1, 2, 3, 4, 5]`).
So I should probably change this to return a `ref T[]`, to match ldc's behavior.

I think I still need `@trusted` for the outer functions as I get this error if I remove `@trusted` on `_d_arrayappendcT`.
```
std/uni.d(6165): Error: @safe function std.uni.Stack!(InversionList!(GcPolicy)).Stack.push cannot call @system function object._d_arrayappendcT!(InversionList!(GcPolicy))._d_arrayappendcT
```